### ISSUE TITLE
Increased visual-diff flakiness during CI

### ISSUE THREAD

**Issue by rsimha** [ampproject/amphtml#32461]
After running `gulp visual-diff` on CI, we have the odd diff show up in Percy. Of late, I'm seeing way more flaky tests. Not sure if this is due to new tests, or if it's an infrastructure issue (new version of Chrome, puppeteer, or the Percy agent?) that's causing this spike in flakiness.

E.g. See [these diffs](https://percy.io/ampproject/amphtml/builds/8716818/changed/495171451?browser=firefox&browser_ids=5&subcategories=unreviewed%2Cchanges_requested%2Capproved&viewLayout=side-by-side&viewMode=new&width=800&widths=375%2C411%2C800%2C1400) for a PR that does not touch runtime code.

**Comment by danielrozenberg** [comment]
On initial investigation this looks related to the `layout-aspect-ratio-css` experiment - it's set to 10% in `prod-config.json` which explains why it produces flakiness in these tests - example diff of [one of the flaky tests](https://percy.io/ampproject/amphtml/builds/8739672/changed/496279614):

```patch
   <h2>responsive - 2x1 ratio - Text Node</h2>
-  <amp-layout layout="responsive" width="2" height="1" class="i-amphtml-element i-amphtml-layout-responsive i-amphtml-layout-size-defined i-amphtml-built i-amphtml-layout" i-amphtml-layout="responsive" style="aspect-ratio: 2 / 1;"><div class="i-amphtml-fill-content">
+  <amp-layout layout="responsive" width="2" height="1" class="i-amphtml-element i-amphtml-layout-responsive i-amphtml-layout-size-defined i-amphtml-built i-amphtml-layout" i-amphtml-layout="responsive"><i-amphtml-sizer slot="i-amphtml-svc" style="padding-top: 50%;"></i-amphtml-sizer><div class="i-amphtml-fill-content">
     Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec tincidunt augue ut dolor rutrum, sit amet tristique velit fringilla. Duis pretium ante dui, a congue erat elementum dictum. Suspendisse sit amet dui libero. Sed sodales quam in mauris aliquet laoreet. Ut iaculis leo ipsum, ac commodo tellus tempus lobortis. Nam molestie semper suscipit. Quisque consectetur purus et tortor mollis porttitor. Morbi tempus sit amet mauris eget mattis. Sed dignissim, quam ut mattis ullamcorper, lorem ex venenatis purus, a sollicitudin enim tortor sit amet justo.
   </div></amp-layout>
 
   <h2>responsive - 2x1 ratio - DIV Node</h2>
-  <amp-layout layout="responsive" width="2" height="1" class="i-amphtml-element i-amphtml-layout-responsive i-amphtml-layout-size-defined i-amphtml-built i-amphtml-layout" i-amphtml-layout="responsive" style="aspect-ratio: 2 / 1;"><div class="i-amphtml-fill-content">
+  <amp-layout layout="responsive" width="2" height="1" class="i-amphtml-element i-amphtml-layout-responsive i-amphtml-layout-size-defined i-amphtml-built i-amphtml-layout" i-amphtml-layout="responsive"><i-amphtml-sizer slot="i-amphtml-svc" style="padding-top: 50%;"></i-amphtml-sizer><div class="i-amphtml-fill-content">
     <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec tincidunt augue ut dolor rutrum, sit amet tristique velit fringilla. Duis pretium ante dui, a congue erat elementum dictum. Suspendisse sit amet dui libero. Sed sodales quam in mauris aliquet laoreet. Ut iaculis leo ipsum, ac commodo tellus tempus lobortis. Nam molestie semper suscipit. Quisque consectetur purus et tortor mollis porttitor. Morbi tempus sit amet mauris eget mattis. Sed dignissim, quam ut mattis ullamcorper, lorem ex venenatis purus, a sollicitudin enim tortor sit amet justo.</div>
   </div></amp-layout>
 
   <h2>responsive - svg - 1x1 ratio </h2>
-  <amp-layout layout="responsive" width="1" height="1" class="i-amphtml-element i-amphtml-layout-responsive i-amphtml-layout-size-defined i-amphtml-built i-amphtml-layout" i-amphtml-layout="responsive" style="aspect-ratio: 1 / 1;"><div class="i-amphtml-fill-content">
+  <amp-layout layout="responsive" width="1" height="1" class="i-amphtml-element i-amphtml-layout-responsive i-amphtml-layout-size-defined i-amphtml-built i-amphtml-layout" i-amphtml-layout="responsive"><i-amphtml-sizer slot="i-amphtml-svc" style="padding-top: 100%;"></i-amphtml-sizer><div class="i-amphtml-fill-content">
     <svg viewBox="0 0 100 100">
```

@powerivq got any ideas what's going on?

**Comment by powerivq** [comment]
@dvoytenko Looks like it, correct?

@danielrozenberg  I wonder how we usually handle this situation: where an experiment affects a visual diff. Should we turn off temporarily till fully deployed?

**Comment by dvoytenko** [comment]
I'll take a look at this one.

**Comment by dvoytenko** [comment]
So. I'm retesting these pages and they all look good. I don't see a regression shown in the visual diff with or without an experiment.

**Comment by honeybadgerdontcare** [comment]
We're doing a validator roll out which changes no runtime or extension javascript and getting visual errors, see https://percy.io/ampproject/amphtml/builds/8742078/changed/496444959?browser=firefox&browser_ids=5&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=800&widths=375%2C411%2C800%2C1400
/cc @MichaelRybak

**Comment by dvoytenko** [comment]
@danielrozenberg is there any way to know for certain which browser is used for snapshotting? E.g. which Chromium version is used?

**Comment by dvoytenko** [comment]
Actually, it looks like all diffs are done via Firefox? Is that configurable at all? Do we know which FF?

**Comment by dvoytenko** [comment]
I have one guess, but it could be completely off. Is it possible that the page snapshots are produced on Chrome, but visual diff is done on FF?

**Comment by dvoytenko** [comment]
@danielrozenberg could you pls comment on the changes here for posterity?

**Comment by danielrozenberg** [comment]
Sure - we've switched the Percy project's config to render on Chrome instead of Firefox. I'll look into running the visual diff tests in parallel on both Chrome and Firefox and report them to two separate Percy projects.
@rsimha wrote #32546 to ensure visual diff tests only run on CircleCI, not on Travis

**Comment by dvoytenko** [comment]
Can we also confirm the current Chrome version running on Percy?

**Comment by rsimha** [comment]
> I have one guess, but it could be completely off. Is it possible that the page snapshots are produced on Chrome, but visual diff is done on FF?

DOM snapshots are created on Chrome latest stable channel (on CircleCI), but as we discovered recently, they are being rendered on the Percy side using an older version of Chrome.

**CI build:** [here](https://app.circleci.com/pipelines/github/ampproject/amphtml/1534/workflows/8ae5ab2f-2a24-41c8-96f2-cbf5d071dfe3/jobs/14666) (expand the last two sections)
**Corresponding Percy build:** [here](https://percy.io/ampproject/amphtml/builds/8762503) (click on the `ⓘ' icon)

![image](https://user-images.githubusercontent.com/26553114/107411501-992acb00-6adc-11eb-85ff-c07df8652b1e.png)

**Comment by danielrozenberg** [comment]
After running for a couple of days, it looks like #32554 fixed the crux of the flakiness, and I removed two tests that were individually flaky with #32686 

I contacted Percy to ask for a timeline for updating their server-side Chrome version used for performing the diffs, will update those involved offline

Closing this issue as this seems resolved - please reopen or create another issue if you encounter more flakiness!
